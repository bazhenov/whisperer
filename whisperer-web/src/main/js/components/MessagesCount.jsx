import React from 'react'

export default ({ totalCount, filteredCount, clearMessages }) => {
	if (totalCount === 0) return null;
	return <div id="messages-records-count">
		<strong>{filteredCount}{filteredCount !== totalCount ? ' (' + totalCount + ')' : ''} records</strong>
		<a href="#" onClick={(e) => { clearMessages(); e.preventDefault() }} id="messages-clear-button" className="btn btn-danger">CLEAR</a>
	</div>;
};