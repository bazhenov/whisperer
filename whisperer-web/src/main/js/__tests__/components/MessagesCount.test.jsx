import React from "react";
import MessagesCount from "../../components/MessagesCount.jsx";
import renderer from "react-test-renderer";

it('MessagesCount', () => {
	let tree = renderer.create(
		<MessagesCount filteredCount={0} totalCount={0} />
	).toJSON();
	expect(tree).toMatchSnapshot();

	tree = renderer.create(
		<MessagesCount filteredCount={10} totalCount={10} />
	).toJSON();
	expect(tree).toMatchSnapshot();

	tree = renderer.create(
		<MessagesCount filteredCount={5} totalCount={10} />
	).toJSON();
	expect(tree).toMatchSnapshot();
});