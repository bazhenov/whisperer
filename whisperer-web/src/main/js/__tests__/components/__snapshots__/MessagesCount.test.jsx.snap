// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesCount 1`] = `null`;

exports[`MessagesCount 2`] = `
<div
  id="messages-records-count"
>
  <strong>
    10
  </strong>
  
   records
</div>
`;

exports[`MessagesCount 3`] = `
<div
  id="messages-records-count"
>
  <strong>
    5
  </strong>
   (10)
   records
</div>
`;
